objects/src/errors.rs

```diff
@@ -22,6 +22,34 @@ use crate::{
     ACCOUNT_UPDATE_MAX_SIZE, MAX_INPUTS_PER_NOTE, MAX_INPUT_NOTES_PER_TX, MAX_OUTPUT_NOTES_PER_TX,
 };
 
+// ACCOUNT COMPONENT TEMPLATE ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum AccountComponentTemplateError {
+    #[error("error creating AccountComponent")]
+    AccountComponentError(#[source] AccountError),
```

Comment:
As mentioned in one of the previous comments, maybe instead of this we should have something like `AccountError::TemplateError` variant.

Reply:
I mentioned in the other comment as well because we were holding an `AccountError` in one of the other variants this was not doable, but after removing the error from the storage map creation function, this was unblocked